File: Leetcode/Easy Problems/1758. Minimum Changes To Make Alternating Binary String/dec-23-2023.cpp
/*
  December 23, 2023
  - Runtime Achieved: 0 ms 
  - Memory Achieved: 7.5 MB
*/
class Solution {
public:
    int minOperations(string s) {
        int len = s.length();
        int n = 0;
        for (int i = 0; i < len; i++) {
            if (i % 2) {
                if (s[i] == '1') {
                    n++;
                }
            } else {
                if (s[i] == '0') {
                    n++;
                }
            }
        }
        return min(len - n, n);
    }   
};

File: Leetcode/Easy Problems/1758. Minimum Changes To Make Alternating Binary String/dec-23-2023.js
/**
  December 23, 2023
  - Runtime Achieved: 53 ms 
  - Memory Achieved: 42.8 MB
/**
 * @param {string} s
 * @return {number}
 */
var minOperations = function(s) {
    let len = s.length;
    let n = 0;
    for(let i = 0; i < len; i++) {
        if(i % 2) {
            if(s[i] === '1')
                n++;
        } 
        else {
            if(s[i] === '0')
                n++;
        }
    }
    return Math.min(len - n, n);
};
